src/thriveopedia/Thriveopedia.cs

```diff
@@ -0,0 +1,338 @@
+    public NodePath PageTreeContainerPath = null!;
+
+    [Export]
+    public NodePath PageTitlePath = null!;
+
+    [Export]
+    public NodePath PageTreePath = null!;
+
+    [Export]
+    public NodePath HomePagePath = null!;
+
+    private TextureButton backButton = null!;
+    private TextureButton forwardButton = null!;
+    private MarginContainer pageContainer = null!;
+    private PanelContainer pageTreeContainer = null!;
+    private Label pageTitle = null!;
+    private Tree pageTree = null!;
+
+    /// <summary>
+    ///   The home page for the Thriveopedia. Keep a special reference so we can return to it easily.
+    /// </summary>
+    private ThriveopediaHomePage homePage = null!;
+
+    /// <summary>
+    ///   Details for the game currently in progress. Null if opened from the main menu.
+    /// </summary>
+    private GameProperties? currentGame;
+
+    /// <summary>
+    ///   The currently open Thriveopedia page.
+    /// </summary>
+    private ThriveopediaPage? selectedPage;
+
+    /// <summary>
+    ///   All Thriveopedia pages and their associated items in the page tree.
+    /// </summary>
+    private Dictionary<ThriveopediaPage, TreeItem> allPages = new();
+
+    /// <summary>
+    ///   Page history for backwards navigation.
+    /// </summary>
+    private Stack<ThriveopediaPage> pageHistory = new();
+
+    /// <summary>
+    ///   Page future for forwards navigation (if the player has already gone backwards).
+    /// </summary>
+    private Stack<ThriveopediaPage> pageFuture = new();
+
+    [Signal]
+    public delegate void OnThriveopediaClosed();
+
+    /// <summary>
+    ///   The currently open Thriveopedia page. Defaults to the home page if none has been set.
+    /// </summary>
+    public ThriveopediaPage SelectedPage
+    {
+        get => selectedPage ?? homePage;
+        set
+        {
+            // Hide the last page and show the new page
+            selectedPage?.Hide();
+
+            selectedPage = value;
+            selectedPage.Show();
+
+            // The home page is always the first in the history, so ignore it
+            backButton.Disabled = pageHistory.Count <= 1;
+            forwardButton.Disabled = pageFuture.Count == 0;
+
+            pageTitle.Text = SelectedPage.TranslatedPageName;
+            SelectInTreeWithoutEvent(selectedPage);
+        }
+    }
+
+    /// <summary>
+    ///   Details for the game currently in progress. Null if opened from the main menu. When set, initialises all
+    ///   pages requiring an active game.
+    /// </summary>
+    public GameProperties? CurrentGame
+    {
+        get => currentGame;
+        set
+        {
+            // We only want to initialise these pages once per game
+            if (value == currentGame)
+                return;
+
+            currentGame = value;
+
+            // Add all pages associated with a game in progress
+            if (currentGame != null)
+            {
+                AddPage("CurrentWorld");
```

Comment:
Related to my earlier comment about "Home" capitalization, is it actually a node name? If so then it is fine as is, and well to avoid my future confusion you could add a comment on that base property.